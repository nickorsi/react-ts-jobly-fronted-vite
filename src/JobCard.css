.card {
  background-image: linear-gradient(hsl(0, 0%, 100%), hsl(0, 0%, 90%));
  border: solid 2px black;
  border-radius: 10px;
  padding: 15px;
  margin: 15px;
}

.job-title {
  font-weight: bold;
}