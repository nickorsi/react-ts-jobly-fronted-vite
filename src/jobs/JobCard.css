.card {
  background-color: white;
  border: solid 2px black;
  border-radius: 10px;
  padding: 15px;
  margin: 15px;
}

.job-pay-info {
  margin-top: 15px;
}

.job-titles p, .job-pay-info p {
  margin-bottom: 0;
}

.job-title {
  font-weight: bold;
}




