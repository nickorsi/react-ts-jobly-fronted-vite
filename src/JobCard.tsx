import { JobDataFromAPI } from "./JobList"
import './JobCard.css';

/**
 * JobCard component that renders information about a specific job.
 *
 * Takes {id, title, companyName, companyHandle salary, equity} as props
 *
 * JobList -> JobCard
 */

function JobCard ({id, title, companyName, salary, equity}: JobDataFromAPI ) {
    return (
        <div id={`${id}`} className="job card">
            <div className="job-titles">
                <p>Job Title: <span className='job-title'>{ title }</span></p>
                <p className='job-company'>Company: { companyName }</p>
            </div>
            <div className="job-pay-info">
                <p className='job-salary'>Salary: { salary }</p>
                <p className='job-equity'>Equity: { equity }</p>
            </div>
        </div>
    )
}

export default JobCard