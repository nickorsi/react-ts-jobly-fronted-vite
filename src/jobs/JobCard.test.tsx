import {describe, expect, it} from "vitest";
import { render } from "@testing-library/react";

import JobCard from "./JobCard.tsx";

const defaultJob = {
    "id": 1,
    "title": "Job 1",
    "salary": 1,
    "equity": "1",
    "companyHandle": "c1",
    "companyName": "Company 1",
}


describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(
            <JobCard
                id={defaultJob.id}
                title={defaultJob.title}
                salary={defaultJob.salary}
                equity={defaultJob.equity}
                companyHandle={defaultJob.companyHandle}
                companyName={defaultJob.companyName}
            />
        );
    });
    it("matches the snapshot", function () {
        const { container } = render(
            <JobCard
                id={defaultJob.id}
                title={defaultJob.title}
                salary={defaultJob.salary}
                equity={defaultJob.equity}
                companyHandle={defaultJob.companyHandle}
                companyName={defaultJob.companyName}
            />
        );

        expect(container).toMatchSnapshot();
    });
});