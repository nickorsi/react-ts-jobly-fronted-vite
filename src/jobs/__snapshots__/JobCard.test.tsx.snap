// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <div
    class="job card"
    id="1"
  >
    <div
      class="job-titles"
    >
      <p>
        Job Title: 
        <span
          class="job-title"
        >
          Job 1
        </span>
      </p>
      <p
        class="job-company"
      >
        Company: 
        Company 1
      </p>
    </div>
    <div
      class="job-pay-info"
    >
      <p
        class="job-salary"
      >
        Salary: 
        1
      </p>
      <p
        class="job-equity"
      >
        Equity: 
        1
      </p>
    </div>
  </div>
</div>
`;
